.github/copilot-instructions.md

```diff
@@ -0,0 +1,25 @@
+## Change Logging
+- Each time you generate code, note the changes in changelog.md
+- Follow semantic versioning guidelines
+- Include date and description of changes
+- periodically perform git commits with clear messages when appropriate
+- Never change the branch that we start with for managablity
+
+## Code Style
+- Follow the existing code style in the repository
+- Use consistent indentation (2 spaces)
+- Follow the BASH style aleady in code base expecally "${variable}" for variables
+- Alays have no extra whitespace at the end or beginning of lines
+- Use pycodestyle for Python code
+
+## Code Quality
+- Ensure code is clean, well-commented, and follows best practices
+- Use consistent naming conventions
+- Avoid unnecessary complexity at all costs and make sure the code is easy to understand by average developers
+- Avoid over-engineering solutions
+- Use readable code that conveys intent and meaning over comments
+- Write unit tests for new features and bug fixes
+- Ensure code is modular and reusable
+
+## Documentation
+- use matlab docstrings for functions and classes
```

Comment:
I don't think we want `matlab` docstrings.

Reply:
Oh yeah numpy duh